### ISSUE TITLE
[1.19.2 Forge Bug]: About the Claw Tools

### ISSUE THREAD

**Issue by SiverDX** [DragonSurvivalTeam/DragonSurvival#324]
### ■ Dragon Survival version and Modpack version (if have)

DragonSurvival-1.19.2-1.5.47.jar

### ■ Single game or server?

Single (Default)

### ■ Do you delete old configs before writing a bug report?

Nope (Default)

### ■ Bug Description and Reproduce

* Pickaxe (and maybe other tools with other harvestable blocks?) don't seem to work with MCreator ores (e. g. from `Gothic RPG` or `Iter RPG`)
* (Some?) special effects (e. g. `Featherite Pickaxe` from `Upgraded Netherite`) don't work

The MCreator part is probably due to the way the block breaking is done, here an example:
``` java
public boolean canHarvestBlock(BlockState state, BlockGetter world, BlockPos pos, Player player) {
    Item var6 = player.getInventory().getSelected().getItem();
    if (var6 instanceof PickaxeItem tieredItem) {
        return tieredItem.getTier().getLevel() >= 2;
    } else {
        return false;
    }
}
```

One option could be:
* Add a config (or datapack tag file) where people can add ores and provide default list (of optional tags)
* In `blockBroken` (`EventHandler.java`) check config and simulate the loot drop yourself for the block

Another option is to write a mixin for each mod to also check the Claw Tools

### ■ Expected behavior

-

**Comment by SiverDX** [comment]
There might be a third, better option:

The `HarvestCheck` event:

``` java
@SubscribeEvent
public static void onBlockHarvested(PlayerEvent.HarvestCheck event) {
	Block block = event.getTargetBlock().getBlock();
	ResourceLocation key = ForgeRegistries.BLOCKS.getKey(block);
	if (someList.contains(key)) {
		// Check if this block should be harvestable
		event.setCanHarvest(true);
	}
}
```

**Comment by SiverDX** [comment]
Fixed in the latest update (DragonSurvival-1.19.2-1.5.50.jar) for 1.19.2
